Code/Tools/ProjectManager/Source/GemRepo/GemRepoAddDialog.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <GemRepo/GemRepoAddDialog.h>
+#include <FormLineEditWidget.h>
+
+#include <QVBoxLayout>
+#include <QLabel>
+#include <QLineEdit.h>
+#include <QPushButton>
+
+namespace O3DE::ProjectManager
+{
+    GemRepoAddDialog::GemRepoAddDialog(QWidget* parent)
+        : QDialog(parent)
+    {
+        setWindowTitle(tr("Add a User Repository"));
+        setModal(true);
+
+        QVBoxLayout* vLayout = new QVBoxLayout();
+        vLayout->setContentsMargins(30, 30, 25, 10);
+        vLayout->setSpacing(0);
+        setLayout(vLayout);
+
+        QLabel* instructionTitleLabel = new QLabel(tr("Enter a valid path to add a new user repository"));
+        instructionTitleLabel->setObjectName("gemRepoAddDialogInstructionTitleLabel");
+        instructionTitleLabel->setAlignment(Qt::AlignLeft);
+        vLayout->addWidget(instructionTitleLabel);
+
+        vLayout->addSpacing(10);
+
+        QLabel* instructionContextLabel = new QLabel(tr("The path can be a Repository URL or a Local Path in your directory."));
+        instructionContextLabel->setAlignment(Qt::AlignLeft);
+        vLayout->addWidget(instructionContextLabel);
+
+        m_repoPath = new FormLineEditWidget(tr("Repository Path"), "", this);
+        m_repoPath->setFixedWidth(500);
+        vLayout->addWidget(m_repoPath);
+
+        vLayout->addSpacing(40);
+
+        QDialogButtonBox* dialogButtons = new QDialogButtonBox();
+        dialogButtons->setObjectName("footer");
+        vLayout->addWidget(dialogButtons);
+
+        QPushButton* cancelButton = dialogButtons->addButton(tr("Cancel"), QDialogButtonBox::RejectRole);
+        cancelButton->setProperty("secondary", true);
+        QPushButton* continueButton = dialogButtons->addButton(tr("Add"), QDialogButtonBox::ApplyRole);
+
+        connect(cancelButton, &QPushButton::clicked, this, &GemRepoAddDialog::CancelButtonPressed);
+        connect(continueButton, &QPushButton::clicked, this, &GemRepoAddDialog::ContinueButtonPressed);
+    }
+
+    QDialogButtonBox::ButtonRole GemRepoAddDialog::GetButtonResult()
+    {
+        return m_buttonResult;
+    }
+
+    QString GemRepoAddDialog::GetRepoPath()
+    {
+        return m_repoPath->lineEdit()->text();
+    }
+
+    void GemRepoAddDialog::CancelButtonPressed()
+    {
+        m_buttonResult = QDialogButtonBox::RejectRole;
+        close();
+    }
+
+    void GemRepoAddDialog::ContinueButtonPressed()
+    {
+        m_buttonResult = QDialogButtonBox::ApplyRole;
+        close();
+    }
```

Comment:
instead of these custom methods, storing and querying the result, I think you can just point your buttons to the built in accept() and reject() QDialog functions and then QDialog::exec() will return that code. 

https://doc.qt.io/qt-5/qdialog.html#DialogCode-enum